swap/src/protocol/alice/message0.rs

```diff
@@ -49,9 +50,11 @@ pub struct Behaviour {
 }
 
 impl Behaviour {
-    pub fn send(&mut self, channel: ResponseChannel<AliceToBob>, msg: Message0) {
+    pub fn send(&mut self, channel: ResponseChannel<AliceToBob>, msg: Message0) -> Result<()> {
         let msg = AliceToBob::Message0(Box::new(msg));
-        self.rr.send_response(channel, msg);
+        self.rr
+            .send_response(channel, msg)
+            .map_err(|_| anyhow!("Sending Amounts response failed"))
```

Comment:
Did you try using `.context` instead? That is usually preferable unless the error returned doesn't implement `std::error::Error`.

```suggestion
            .context("Sending Amounts response failed")
```

Reply:
> Did you try using `.context` instead? That is usually preferable unless the error returned doesn't implement `std::error::Error`.

I don't think you can use context here because the error is an `AliceToBob` enum (the message that failed to send).